product-service/handler.js

```diff
@@ -0,0 +1,145 @@
+const AWS = require("aws-sdk");
+const generateUniqueId = require("./src/utils/generateUniqueId");
+
+AWS.config.update({ region: "eu-west-1" });
+const dynamodb = new AWS.DynamoDB.DocumentClient();
+
+module.exports.getProductsList = async () => {
+  try {
+    const productsParams = {
+      TableName: process.env.PRODUCTS_TABLE,
+    };
+
+    const stocksParams = {
+      TableName: process.env.STOCKS_TABLE,
+    };
+
+    const [productsData, stocksData] = await Promise.all([
+      dynamodb.scan(productsParams).promise(),
+      dynamodb.scan(stocksParams).promise(),
+    ]);
+
+    const productsWithStocks = productsData.Items.map((product) => {
+      const stockInfo = stocksData.Items.find(
+        (item) => item.product_id === product.id
+      );
+      return {
+        ...product,
+        count: stockInfo ? stockInfo.count : 0,
```

Comment:
You're assuming count will exist on stockInfo if stockInfo exists.
You could do something safer like
```suggestion
        count: stockInfo?.count ?? 0,
```

Then again, if the intent is for this code to throw if for some reason you get stockInfo back but it doesn't have a count property (such as if the data schema has changed), this is fine

Reply:
I will go with the option you've suggested, as it's safer.